watcher/src/watcher_db.rs

```diff
@@ -901,6 +906,68 @@ impl WatcherDB {
             Err(err) => Err(err.into()),
         }
     }
+
+    /// Remove all the data associated with a given source url.
+    pub fn remove_all_for_source_url(&self, src_url: &Url) -> Result<(), WatcherDBError> {
+        if !self.write_allowed {
+            //return Err(WatcherDBError::ReadOnly);
```

Comment:
Should something happen here?

Reply:
Good catch! I commented it out for some debugging I was doing.